### PR TITLE
CDAP-17078 consolidate multiple outputs from same sink

### PR DESCRIPTION
**Pull request by albertshau**
Some sinks have multiple outputs, which can cause a lot of
data recomputation. Added logic to consolidate these outputs
when stage consolidation is enabled.

### CODE REVIEW

File: cdap-app-templates/cdap-etl/hydrator-spark-core-base/src/main/java/io/cdap/cdap/etl/spark/batch/SparkBatchSinkFactory.java [hunk 1]
```diff
@@ -173,9 +165,34 @@ private void addOutput(String stageName, String datasetName, String alias, Map<S
         sec.saveAsDataset(rdd, datasetInfo.getDatasetName(), datasetInfo.getDatasetArgs());
       }
     }
+
+    if (outputFormats.size() == 1) {
+      saveUsingOutputFormat(outputFormats.values().iterator().next(), rdd);
+    }
```
Comment (MEseifan) [inline cdap-app-templates/cdap-etl/hydrator-spark-core-base/src/main/java/io/cdap/cdap/etl/spark/batch/SparkBatchSinkFactory.java]: Shouldn't we return after calling `saveUsingOutputFormat`?

**Review by MEseifan — APPROVED** [review]
One minor comment, otherwise LGTM

File: cdap-app-templates/cdap-etl/hydrator-spark-core-base/src/main/java/io/cdap/cdap/etl/spark/batch/SparkBatchSinkFactory.java (hunk 1 above)
Comment (albertshau) [reply cdap-app-templates/cdap-etl/hydrator-spark-core-base/src/main/java/io/cdap/cdap/etl/spark/batch/SparkBatchSinkFactory.java]: agh yes

**Comment by albertshau** [comment]
https://builds.cask.co/browse/CDAP-DUT7306

**Pull request merged**